game-core/src/main/java/games/strategy/triplea/ui/screen/UnitsDrawer.java

```diff
@@ -110,14 +111,22 @@ public void draw(
       throw new IllegalStateException("Type not found:" + unitType);
     }
     final GamePlayer owner = data.getPlayerList().getPlayerId(playerName);
+    final boolean damagedImage = damaged > 0 || bombingUnitDamage > 0;
+    final boolean disabledImage = disabled;
     final Optional<Image> img =
-        uiContext
-            .getUnitImageFactory()
-            .getImage(type, owner, damaged > 0 || bombingUnitDamage > 0, disabled);
+        uiContext.getUnitImageFactory().getImage(type, owner, damagedImage, disabledImage);
 
     if (img.isEmpty() && !uiContext.isShutDown()) {
+      final List<String> imageQualifiers = new ArrayList<>();
+      if (damagedImage) {
+        imageQualifiers.add("damaged ");
+      }
+      if (disabledImage) {
+        imageQualifiers.add("disabled ");
```

Comment:
We can have, or even require sometimes a disabled & damaged image? I kinda thought they were mutually exclusive. If not, it implies a pretty large number of image permutations that is probably not typically included (which would be kinda bad news).

Reply:
Ah, looks like I missed the "else" in the UnitImageFactory.  I somehow misread it as two `if` statements instead of a `if else if`.